flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -537,7 +543,6 @@ public YarnClusterClient createCluster(
 		Preconditions.checkNotNull(userJarFiles, "User jar files should not be null.");
 
 		AbstractYarnClusterDescriptor yarnClusterDescriptor = createDescriptor(applicationName, cmdLine);
-		yarnClusterDescriptor.setFlinkConfiguration(config);
```

Comment:
You should give this configuration to the `createDescriptor`  method where it is used to instantiate the `YarnClusterDescriptor`.